src/client/sandbox/cookie/index.js

```diff
@@ -127,7 +130,7 @@ export default class CookieSandbox extends SandboxBase {
 
         // NOTE: First, update our client cookies cache with a client-validated cookie string,
         // so that sync code can immediately access cookies.
-        const parsedCookie = cookieUtils.parse(value);
+        const parsedCookie = typeof value === 'string' ? cookieUtils.parse(value) : value;
```

Comment:
Add test for this change

Reply:
All client tests which test the server synchronization go through this condition.